crates/router/src/connector/iatapay/transformers.rs

```diff
@@ -0,0 +1,369 @@
+            }),
+            ..item.data
+        })
+    }
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct RedirectUrls {
+    success_url: String,
+    failure_url: String,
+}
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct IatapayPaymentsRequest {
+    merchant_id: String,
+    amount: i64,
+    currency: String,
+    country: String,
+    locale: String,
+    redirect_urls: RedirectUrls,
+    notification_url: String,
+}
+
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct IatapayCard {
+    name: Secret<String>,
+    number: Secret<String, common_utils::pii::CardNumber>,
+    expiry_month: Secret<String>,
+    expiry_year: Secret<String>,
+    cvc: Secret<String>,
+    complete: bool,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for IatapayPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let country = item.get_billing_country()?.to_string();
+        let payload = Self {
+            merchant_id: IatapayAuthType::try_from(&item.connector_auth_type)?.merchant_id,
+            amount: item.request.amount,
+            currency: item.request.currency.to_string(),
+            country: country.clone(),
+            locale: format!("en-{}", country),
+            redirect_urls: get_redirect_url(item),
+            notification_url: item.request.get_webhook_url()?,
+        };
+        Ok(payload)
+    }
+}
+
+fn get_redirect_url(item: &types::PaymentsAuthorizeRouterData) -> RedirectUrls {
+    let return_url = item
+        .request
+        .get_return_url()
+        .unwrap_or("https://hyperswitch.io".to_string());
+    RedirectUrls {
+        success_url: return_url.clone(),
+        failure_url: return_url,
+    }
+}
+
+//TODO: Fill the struct with respective fields
+// Auth Struct
+pub struct IatapayAuthType {
+    pub(super) client_id: String,
+    pub(super) merchant_id: String,
+    pub(super) client_secret: String,
+}
+
+impl TryFrom<&types::ConnectorAuthType> for IatapayAuthType {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        match auth_type {
+            types::ConnectorAuthType::SignatureKey {
+                api_key,
+                key1,
+                api_secret,
+            } => Ok(Self {
+                client_id: api_key.to_string(),
+                merchant_id: key1.to_string(),
+                client_secret: api_secret.to_string(),
+            }),
+            _ => Err(errors::ConnectorError::FailedToObtainAuthType)?,
+        }
+    }
+}
+// PaymentsResponse
+//TODO: Append the remaining status flags
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
```

Comment:
do you need all here? 
Debug, Serialize is enough for `payment_request` structs
Debug, Deserialize is enough for `payment_response` structs
add Default, Clone, PartialEq etc only if needed